### PR TITLE
VC Issuer Guide for tbDEX

### PR DESCRIPTION
**Pull request by chris-tbd**
## What type of PR is this? (check all applicable)

- [x] 📝 Documentation Update

## Description

This PR adds two Issuer guides:
* [Issuer Overview](https://deploy-preview-1111--tbd-website-developer.netlify.app/docs/tbdex/issuer/overview)
* [Credential Issuance](https://deploy-preview-1111--tbd-website-developer.netlify.app/docs/tbdex/issuer/vc-issuance)

## Added code snippets?
- [x] 👍 yes
- [ ] 🙅 no, because they aren't needed

## Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [x] 🙋 no, because I need help

### No tests? Add a note
Need help getting tests up and running for these

## Added to documentation?

- [ ] 📜 readme
- [ ] 📜 contributing.md
- [x] 📓 general documentation
- [ ] 🙅 no documentation needed

### CODE REVIEW

File: site/docs/tbdex/issuer/quickstart.mdx [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+
+class SanctionsCredential {
+  constructor(listsChecked) {
+    // indicates the sanctions lists checked against
+    this.listsChecked = listsChecked; 
+  }
+}
+```
+
+In our example above, we can offer a class property that stores all the sanctions lists checked against to issue the credential. You have wide flexibility with what you can store in your credential class, which means that you're free to include any public attributes that may be relevant to the credential you are verifying.
```
Comment (angiejones) [inline site/docs/tbdex/issuer/quickstart.mdx:77]:
```suggestion
In the example above, you provide a class property that stores all the sanctions lists checked against to issue the credential. You have wide flexibility with what you can store in the credential model class, which means that you're free to include any public attributes that may be relevant to the credential you are issuing.
```

File: site/docs/tbdex/issuer/quickstart.mdx [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+Your credential model is ** not ** the place to store private information about the user you are issuing the credential to, such as banking information or other PII.
```
Comment (angiejones) [inline site/docs/tbdex/issuer/quickstart.mdx:79]: do you mean the model class or the credential itself?

File: site/docs/tbdex/issuer/quickstart.mdx [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+
+In our example above, we can offer a class property that stores all the sanctions lists checked against to issue the credential. You have wide flexibility with what you can store in your credential class, which means that you're free to include any public attributes that may be relevant to the credential you are verifying.
+
+Your credential model is ** not ** the place to store private information about the user you are issuing the credential to, such as banking information or other PII. 
+
+## Implement Issuance
+
+With your web server structure in place and credential designed, you're now ready to build out your issuance codepath. 
+
+** BLOCKED ON THIS PART TO SHOW HOW TO ACTUALLY TAKE IN INFO TO BE VERIFIED **
```
Comment (angiejones) [inline site/docs/tbdex/issuer/quickstart.mdx:85]: I think there's some misunderstanding here. I've removed the word "verify" a couple of times throughout the guide. A verifier has a very distinct meaning in the context of VCs, and is a different entity than the issuer. A verifier is someone you give an issued credential to.

The issuer is the one making claims. They wouldn't necessarily take any info and verify it. They'd do their own checks (check a real Sanctions list, or check an internal system to make sure the person is an employee, or actually earned a degree from their university). The issuer is the one qualified to make to the claim

Comment (angiejones) [reply site/docs/tbdex/issuer/quickstart.mdx:85]: Now what you might want to highlight is the act of a applying for or requesting a credential from an issuer

File: site/docs/tbdex/issuer/quickstart.mdx [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+        // Process & verify user data
+        const listsChecked = verifyUserData(payload);
+
+        if (listsChecked == null) {
+            res.status(401).json({
+                errors: "User failed sanctions Check."
+            })
+        }
+
+        // Mint the credential
```
Comment (angiejones) [inline site/docs/tbdex/issuer/quickstart.mdx:111]:
```suggestion
        // Create the credential
```

File: site/docs/tbdex/issuer/quickstart.mdx [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+            type: "SanctionsCredential",
+            issuer: issuerDid.did,
+            subject: signer,
+            data: new SanctionsCredential(listsChecked)
+        });
+    }
+ });
+```
+
+Now the `vc` object you have at the end of this snippet will be a raw VC that looks like this:
```
Comment (angiejones) [inline site/docs/tbdex/issuer/quickstart.mdx:122]:
```suggestion
The `vc` object that's returned will be a raw VC that looks like this:
```

File: site/docs/tbdex/issuer/quickstart.mdx [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+  issuer: "did:key:z6Mkn4w6nSaWe4fjNJRvaHZwFnMm5VexvjzDeozEu2G7jC34",
+  issuanceDate: "2023-12-14T08:20:41Z",
+  credentialSubject: {
+    id: "did:key:z6MkjGSeekPGE9QfczHWyW8v2ZzJU68kqSHzV7L2dmQyuyDu",
+    listsChecked: "FBI's Most Wanted, USA Watchlist, EU Watchlist"
+  }
+}
+```
+
+To complete your basic VC issuance flow, you can now sign and return the credential. Signing will return a [VC JSON Web Token](/docs/web5/build/verifiable-credentials/jwt-to-vc#what-is-a-jwt), which is ideal for secure transmission of the credential:
```
Comment (angiejones) [inline site/docs/tbdex/issuer/quickstart.mdx:138]:
```suggestion
To complete the issuance, you must sign the credential. Signing will return a [VC JSON Web Token](/docs/web5/build/verifiable-credentials/jwt-to-vc#what-is-a-jwt), which is ideal for secure transmission of the credential:
```

File: site/docs/tbdex/issuer/quickstart.mdx [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+
+```typescript
+const signedVC = await vc.sign({did: issuerDid});
+return res.status(201).json({ credential: signedVC });
+```
+
+With that, you've implmented a basic API service to accept requests, process user data, and issue a VC!
```
Comment (angiejones) [inline site/docs/tbdex/issuer/quickstart.mdx:145]:
```suggestion
With that, you've implmented a basic credential issuance service to accept requests and issue VCs.
```

File: site/docs/tbdex/issuer/quickstart.mdx [hunk 8]
```diff
@@ -0,0 +1,159 @@
...
+
+## Additional Resources
```
Comment (angiejones) [inline site/docs/tbdex/issuer/quickstart.mdx:147]: let's delete this section

**Review by angiejones — CHANGES_REQUESTED** [review]

File: site/docs/tbdex/issuer/quickstart.mdx [hunk 9]
```diff
@@ -0,0 +1,159 @@
+# Onboarding as a tbDEX Issuer
+
+Issuers are a critical component of the tbDEX ecosystem that ensure trust between Wallets and PFIs by independently verifying claims about the party to whom they are issuing the credential; a typical example of a claim in tbDEX is verifying that a Wallet owner is not on a sanctions list. Although issuance is not a concept that is exclusive to tbDEX - it can be applied across [Web5](/docs/) in general -, as a tbDEX issuer you'll apply the same concepts to your app. The goal of this guide is to serve as a map to existing documentation we've provided for developers using our Web5 SDK, as well as provide you with information specific to your use case as a tbDEX developer. In this guide, we'll cover a basic implementation of an express server using the `@web5/credentials` class to issue your own VC.
+
+## Background
```
[the remark this answers is not in the document]
Comment (chris-tbd) [reply site/docs/tbdex/issuer/quickstart.mdx:11]: I believe your other comments overlapped with this so we should be good to resolve this one

File: site/docs/tbdex/issuer/quickstart.mdx (hunk 8 above)
Comment (chris-tbd) [reply site/docs/tbdex/issuer/quickstart.mdx:160]: Are you referring to the "Additional Resources" Section?

File: site/docs/tbdex/issuer/quickstart.mdx [hunk 10]
```diff
@@ -0,0 +1,159 @@
...
+You'll perform the following commands to install the packages required for your Issuer:
+
+```bash
+npm i @web5/credentials
+npm i express
+```
+
+Within your files, you'll use these imports:
+
+```typescript
```
[the remark this answers is not in the document]
Comment (chris-tbd) [reply site/docs/tbdex/issuer/quickstart.mdx]: Updated

File: site/docs/tbdex/issuer/quickstart.mdx [hunk 11]
```diff
@@ -0,0 +1,159 @@
...
+import { VerifiableCredential } from "@web5/credentials";
+import { api } from './api.js'
+```
+
+Additionally, you'll want to ensure that you create a Decentralized Identifier (DID), which will serve as your Issuer's identity, following the [instructions outlined here](/docs/web5/build/decentralized-identifiers/how-to-create-did#create-a-did-manually). You can find a sample code snippet below:
+
+```typescript
```
[the remark this answers is not in the document]
Comment (chris-tbd) [reply site/docs/tbdex/issuer/quickstart.mdx]: Removed

File: site/docs/tbdex/issuer/quickstart.mdx [hunk 12]
```diff
@@ -0,0 +1,159 @@
...
+import { DidKeyMethod } from '@web5/dids';
+
+const issuerDid = await DidKeyMethod.create();
+```
+
+Note that the method above does require installing the `@web5/dids` package.
```
[the remark this answers is not in the document]
Comment (chris-tbd) [reply site/docs/tbdex/issuer/quickstart.mdx]: Removed

File: site/docs/tbdex/issuer/quickstart.mdx (hunk 2 above)
Comment (chris-tbd) [reply site/docs/tbdex/issuer/quickstart.mdx]: Updated with clarifying text

File: site/docs/tbdex/issuer/quickstart.mdx (hunk 3 above)
Comment (chris-tbd) [reply site/docs/tbdex/issuer/quickstart.mdx]: Thanks for clarifying - I updated the language to be more clear about the difference between verify and check, and this should help devs understand it.

File: site/docs/tbdex/issuer/quickstart.mdx (hunk 8 above)
Comment (angiejones) [reply site/docs/tbdex/issuer/quickstart.mdx:150]: yes

File: site/docs/tbdex/issuer/quickstart.mdx
Comment (angiejones) [inline site/docs/tbdex/issuer/quickstart.mdx:1]: can you rename this file to `onboarding.mdx`?

File: site/docs/tbdex/issuer/quickstart.mdx (hunk 9 above)
[the remark this answers is not in the document]
Comment (angiejones) [reply site/docs/tbdex/issuer/quickstart.mdx:11]: looking at the deploy preview, this section doesn't add any additional value and can be removed completely

https://deploy-preview-1111--tbd-website-developer.netlify.app/docs/tbdex/issuer/quickstart#background

File: site/docs/tbdex/issuer/quickstart.mdx [hunk 13]
```diff
@@ -0,0 +1,162 @@
...
+
+## Background
+
+Although you may know you want to build a service to issue Verifiable Credentails (VCs) to faciliate trust and transactions through tbDEX, you may be looking for more background knowledge before beginning development. Our [overview guide on VCs](/docs/web5/learn/verifiable-credentials) serves as background on the overall topic of issuance and VCs.
+
+Additionally, developing an Issuer app will require use of our `@web5/credentials` package via `npm`, which we'll cover in more specifics later.
+
+## Overview
+
+A credential issuer app runs as an API service that exposes endpoints where third parties can request credentials. You can see a full end-to-end implementation of a credential issuer's workflow [in our documentation here](/docs/web5/build/verifiable-credentials/fan-club-vc). For the sake of this section, we'll go over what you as a tbDEX issuer will need to set up your app:
```
Comment (angiejones) [inline site/docs/tbdex/issuer/quickstart.mdx:19]: this is still pointing to the Taylor Swift example. Please update this to point to the generic VC Issuance guide instead

File: site/docs/tbdex/issuer/quickstart.mdx [hunk 14]
```diff
@@ -0,0 +1,159 @@
...
+
+```typescript
+import { DidKeyMethod } from '@web5/dids';
+
+const issuerDid = await DidKeyMethod.create();
+```
+
+Note that the method above does require installing the `@web5/dids` package.
+
+With all of this, you can start up your basic express API service in a `main.ts` file that looks as follows:
```
[the remark this answers is not in the document]
Comment (angiejones) [reply site/docs/tbdex/issuer/quickstart.mdx]: couple of unanswered questions here

File: site/docs/tbdex/issuer/quickstart.mdx (hunk 2 above)
Comment (angiejones) [reply site/docs/tbdex/issuer/quickstart.mdx]: ok this is not true. in fact, we're specifically gonna need KYC credentials that will store private PII info. I recommend just removing this sentence altogether.

File: site/docs/tbdex/issuer/quickstart.mdx (hunk 13 above)
Comment (angiejones) [reply site/docs/tbdex/issuer/quickstart.mdx:19]: I was pretty sure I wrote a suggestion for this paragraph but now I cant find it 😫 let me try again...

```suggestion
A Credential Issuer app runs as an API service that exposes endpoints where third parties can request credentials. 

If you already have an Issuer app up and running,  check out the guide on [how to issue credentials with Web5](/docs/web5/build/verifiable-credentials/vc-issuance/).

If you need to create a Credential Issuer app from scratch, continue on with this guide where we'll cover how to: 
```

File: site/docs/tbdex/issuer/quickstart.mdx [hunk 15]
```diff
@@ -0,0 +1,162 @@
...
+
+Although you may know you want to build a service to issue Verifiable Credentails (VCs) to faciliate trust and transactions through tbDEX, you may be looking for more background knowledge before beginning development. Our [overview guide on VCs](/docs/web5/learn/verifiable-credentials) serves as background on the overall topic of issuance and VCs.
+
+Additionally, developing an Issuer app will require use of our `@web5/credentials` package via `npm`, which we'll cover in more specifics later.
+
+## Overview
+
+A credential issuer app runs as an API service that exposes endpoints where third parties can request credentials. You can see a full end-to-end implementation of a credential issuer's workflow [in our documentation here](/docs/web5/build/verifiable-credentials/fan-club-vc). For the sake of this section, we'll go over what you as a tbDEX issuer will need to set up your app:
+
+- Create a web server using a technology like ** [express](https://expressjs.com/) ** that will serve as an API endpoint.
```
Comment (angiejones) [inline site/docs/tbdex/issuer/quickstart.mdx:21]:
```suggestion
- Create a web server using a technology like ** [Express](https://expressjs.com/) ** that will serve as an API endpoint.
```

File: site/docs/tbdex/issuer/quickstart.mdx [hunk 16]
```diff
@@ -0,0 +1,162 @@
...
+- Design the credential you'd like to offer.
+- Implement your issuance path, where you'll accept a request for issuance, perform checks required to verify the requester's information, and mint the credential.
```
Comment (angiejones) [inline site/docs/tbdex/issuer/quickstart.mdx:23]:
```suggestion
- Implement an issuance path where you'll accept a request for issuance, perform any necessary checks, and issue the credential.
```

File: site/docs/tbdex/issuer/quickstart.mdx [hunk 17]
```diff
@@ -0,0 +1,162 @@
...
+        }
+
+        const [_, compactJwt] = authHeader.split('Bearer ');
+        const { signer, payload } = await JwtVerifier.verify(compactJwt);
+
+        // Process & verify user data
+        const listsChecked = verifyUserData(payload);
+
+        if (listsChecked == null) {
+            res.status(401).json({
```
Comment (angiejones) [inline site/docs/tbdex/issuer/quickstart.mdx:109]: should this be a return statement? otherwise, it'll still create the credential after the fail, right?

```suggestion
            return res.status(401).json({
```

File: site/docs/tbdex/issuer/quickstart.mdx [hunk 18]
```diff
@@ -0,0 +1,162 @@
...
+
+You'll also need to install `@web5/credentials` to create and issue credentials:
+
+```bash
+npm i @web5/credentials
+```
+
+Within your service's entry point file (e.g. index.js), import the following:
+
+```typescript title=index.ts
```
Comment (angiejones) [inline site/docs/tbdex/issuer/quickstart.mdx:38]: should this be `main`? that's why I asked if `main.ts` is your main entry point in that other comment. wanted to make sure I'm not introducing a new file (index) if it should be main

```suggestion
```typescript title=main.ts
```

File: site/docs/tbdex/issuer/quickstart.mdx [hunk 19]
```diff
@@ -0,0 +1,162 @@
...
+
+To set up the VC issuance service, we'll use Express. Follow these instructions to [install Express](https://expressjs.com/en/starter/installing.html). 
+
+You'll also need to install `@web5/credentials` to create and issue credentials:
+
+```bash
+npm i @web5/credentials
+```
+
+Within your service's entry point file (e.g. index.js), import the following:
```
Comment (angiejones) [inline site/docs/tbdex/issuer/quickstart.mdx:36]:
```suggestion
Within your service's entry point file (e.g. main.ts), import the following:
```

**Review by angiejones — CHANGES_REQUESTED** [review]

File: site/docs/tbdex/issuer/quickstart.mdx (hunk 8 above)
Comment (chris-tbd) [reply site/docs/tbdex/issuer/quickstart.mdx]: Removed

File: site/docs/tbdex/issuer/quickstart.mdx (hunk 2 above)
Comment (chris-tbd) [reply site/docs/tbdex/issuer/quickstart.mdx]: Going to remove - just so I can better understand, how is it that credentials can store PII without exposing that? If the credential is being passed around don't we want to remove PII and just let the VC stand for the claim?

File: site/docs/tbdex/issuer/quickstart.mdx (hunk 14 above)
[the remark this answers is not in the document]
Comment (chris-tbd) [reply site/docs/tbdex/issuer/quickstart.mdx]: Updated all to main.js (switched of ts to js as well)

**Review by angiejones — APPROVED** [review]

File: site/docs/tbdex/issuer/vc-issuance.mdx [hunk 20]
```diff
@@ -0,0 +1,185 @@
...
+            });
+        }
+        const compactJwt = tokenParts[1];
+
+        let signer, payload;
+        try {
+            const verificationResult = await Jwt.verify({ jwt: myJwt });
+            
+            //assuming the user signed the JWT, this is their DID
+            signer = verificationResult.signer;
```
Comment (angiejones) [inline site/docs/tbdex/issuer/vc-issuance.mdx:124]: should this be the `sub` property from the JWT that was passed in?

Comment (chris-tbd) [reply site/docs/tbdex/issuer/vc-issuance.mdx:124]: I was referencing this implementation here https://github.com/TBD54566975/workshop-tbdex-abc-vc-issuer/blob/main/src/api.ts

Is that reference out of date? I'm not too familiar with grabbing this info out of the JWT so want to double check

Comment (angiejones) [reply site/docs/tbdex/issuer/vc-issuance.mdx:124]: that one makes a safe assumption that the person who signed the credential is also the subject of the credential (a self-issued VC). that was for the workshop use case. For a generic use case, that's not a safe assumption. It would be safer to use the subject of the jwt as the subject of the VC you're about to issue
